### PR TITLE
JWT Endpoint Authentication

### PR DESCRIPTION
**Pull request by TayGov**
# Description

Protect all endpoints with bearer token

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Unit test and postman tests

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

SPDBCSC-258

### CODE REVIEW

File: src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+import ca.bc.gov.open.ecrc.security.JWTAuthorizationFilter;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.AnonymousAuthenticationFilter;
```
Comment (sivakaruna) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java]: Are these imports used anywhere?

File: src/ecrc-api/pom.xml [hunk 2]
```diff
@@ -71,7 +71,11 @@
 			<artifactId>jaxb-impl</artifactId>
 			<version>2.2.11</version>
 		</dependency>
-
+		<dependency>
+			<groupId>io.jsonwebtoken</groupId>
+			<artifactId>jjwt</artifactId>
+			<version>0.9.1</version>
```
Comment (shaunmillargov) [inline src/ecrc-api/pom.xml]: Possibly break out version into POM properties?

File: src/ecrc-api/src/main/resources/application.properties [hunk 3]
```diff
@@ -16,6 +16,12 @@ info.app.name=@project.name@
 info.app.description=@project.description@
 info.app.version=@project.version@
 
+#JWT Configs
+ecrc.jwt-header=${ECRC_JWT_HEADER}
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/resources/application.properties]: Would these require an associated change to the properties file?

File: src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java (hunk 1 above)
Comment (TayGov) [reply src/ecrc-api/src/main/java/ca/bc/gov/open/EcrcApiApplication.java]: No I have removed them.

File: src/ecrc-api/pom.xml (hunk 2 above)
Comment (TayGov) [reply src/ecrc-api/pom.xml]: This is something we can discuss as I'm not sure what the best practice is.

File: src/ecrc-api/src/main/resources/application.properties (hunk 3 above)
Comment (TayGov) [reply src/ecrc-api/src/main/resources/application.properties]: Yes they are setup in the properties config.

**Pull request merged**